### ISSUE TITLE
Inconsistent text output in Terminal vs. Text Editor    

### ISSUE THREAD

**Issue by NonStandardModel** [SerenityOS/serenity#7392]
Executing `echo "t\est"` in Terminal will print only _tt_ to output. If we redirect the command output to a file and open it with Text Editor it will show _t st_, while printing the file in terminal (using `cat`) it will still show only _tt_.

Screenshot:
![Issue_InconsistantTextOutput](https://user-images.githubusercontent.com/33322918/119231442-3ad0a800-bb21-11eb-9056-d8f64d2944a7.jpg)

**Comment by alimpfard** [comment]
Terminals and text editors are very distinct in the way they display text, it doesn't make much sense for one to do what the other does (the option to interpret vt100 escapes in the text editor would be interesting, but it would just be an option)

**Comment by NonStandardModel** [comment]
Makes sense. I was comparing the output with what bash or sh in Ubuntu makes. It seems that the interpreter in Serenity is different (but that is just design decision). I am not fluent with vt100 details. For example one other oddity is if you try to do `touch "t\est.txt"`. `ls` will show as filename `t.txttt.txt`.

**Comment by bcoles** [comment]
I would expect TextEditor and Terminal/Shell to render escape codes differently.

> For example one other oddity is if you try to do `touch "t\est.txt"`. `ls` will show as filename `t.txttt.txt`.

That seems like a separate issue (similar to #7377).

![image](https://user-images.githubusercontent.com/434827/119246086-cdc11f00-bbc1-11eb-8d0f-171788e70a01.png)

**Comment by bcoles** [comment]
The second issue is legitimate, but a duplicate.

The first issue in the original post is potentially valid.

TextEditor will preserve the character when saving a document without corruption (and at least in the case of `\e`). It is a question of how these characters should be rendered. A more accurate issue title would be `TextEditor: non-printable chracters are rendered as whitespace`. 

There is no consensus as yet as to how escape characters and non-printable characters should be rendered in the GUI or text widgets. See: https://github.com/SerenityOS/serenity/issues/7279#issuecomment-844370723
